httpx/utils.py

```diff
@@ -256,6 +260,23 @@ def unquote(value: str) -> str:
     return value[1:-1] if value[0] == value[-1] == '"' else value
 
 
+K = typing.TypeVar("K")
+V = typing.TypeVar("V")
+
+
+def flatten_on_values(
+    items: typing.Iterable[
+        typing.Union[typing.Tuple[K, typing.Union[V, typing.Sequence[V]]]]
+    ]
+) -> typing.Iterator[typing.Tuple[K, V]]:
+    for k, v in items:
+        if isinstance(v, collections.abc.Sequence) and not isinstance(v, (str, bytes)):
+            for u in v:
+                yield (k, u)
+        else:
+            yield (k, typing.cast(V, v))
```

Comment:
@tomchristie Here's what the double list comprehension turned into. Let me know if this sounds reasonable to keep in this firm.

Reply:
I'll also note that I like the `(str, bytes)` defensiveness here, otherwise you'd get the actual integer values of the bytes which would be quite strange behavior if a user were to accidentally put bytes into a param.